src/app/script/events_class.cpp

```diff
@@ -213,8 +222,11 @@ class AppEvents : public Events
 
   void onRemoveLastListener(EventType eventType) override {
     switch (eventType) {
+      case BeforeSiteChange:
+        [[fallthrough]];
       case SiteChange:
         App::instance()->context()->remove_observer(this);
+        m_addedObserver = false;
```

Comment:
@dacap Question on how this might work: I added the `m_addedObserver` boolean to avoid doubling app on observers for SiteChange/BeforeSiteChange, but for removing the listener I'm assuming `onRemoveLastListener` would delete the observer if i do something like:

> Listen to sitechanged -> Listen to beforesitechanged -> Remove sitechanged

I think with the code how it is right now this will remove the observer, making the remaining beforesitechange event unable to trigger, what's the best way to solve this? Just keeping track of the fact that I may be listening to both events and only remove the observer when I'm listening to 0?

Reply:
Interesting, probably this `m_addedObserver` can be a counter, so when it starts/goes back to `m_addedObserver==0` we add/remove the observer.